python/dolfinx/io/ngsio.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (C) 2022 Jørgen S. Dokken
+#
+# This file is part of DOLFINx (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+"""Tools to extract data from Gmsh models"""
+import typing
+
+import numpy as np
+import numpy.typing as npt
+
+import basix
+import basix.ufl
+import ufl
+from dolfinx import cpp as _cpp
+from dolfinx.cpp.graph import AdjacencyList_int32
+from dolfinx.mesh import (CellType, GhostMode, Mesh, create_cell_partitioner,
+                          create_mesh, meshtags, meshtags_from_entities)
+
+from mpi4py import MPI as _MPI
+
+__all__ = ["cell_perm_array", "ufl_mesh"]
+
+try:
+    import ngsolve as ngs
+    from ngsPETSc.plex import DMPlexMapping
```

Comment:
What are the installation intructions for `ngsPETSc`?

Reply:
https://ngspetsc.readthedocs.io/en/latest/ngsPETSc.html#installation